src/Microsoft.PowerShell.Commands.Utility/commands/utility/Group-Object.cs

```diff
@@ -518,7 +518,16 @@ protected override void EndProcessing()
             {
                 if (AsHashTable)
                 {
-                    Hashtable hashtable = CollectionsUtil.CreateCaseInsensitiveHashtable();
+                    Hashtable hashtable;
+                    if (CaseSensitive.IsPresent)
+                    {
+                        hashtable = new Hashtable();
+                    }
+                    else
+                    {
+                        hashtable = CollectionsUtil.CreateCaseInsensitiveHashtable();
```

Comment:
I think we can simplify the code and this helps us to add Culture parameter later #9348:
```c#
var comparer = CaseSensitive.IsPresent ? StringComparer.CurrentCulture : StringComparer.CurrentCultureIgnoreCase;
hashtable = Hashtable(comparer);

Reply:
Can do! I'm not sure if these methods use currentculture or ordinal though. I'll check :)